CMakeLists.txt

```diff
@@ -249,22 +250,31 @@ endif (SQLITECPP_BUILD_EXAMPLES)
 option(SQLITECPP_BUILD_TESTS "Build and run tests." OFF)
 if (SQLITECPP_BUILD_TESTS)
     # deactivate some warnings for compiling the gtest library
-    if (NOT MSVC)
+    if (MSVC)
+        set(LIB_SUFFIX ".lib")
+    else ()
         add_compile_options(-Wno-variadic-macros -Wno-long-long -Wno-switch-enum -Wno-float-equal -Wno-conversion-null -Wno-switch-default -Wno-pedantic)
-    endif (NOT MSVC)
-
-    # add the subdirectory containing the CMakeLists.txt for the gtest library
-    # TODO: under Linux, uses libgtest-dev if found
-    # TODO: move to the new googletest Github repository
-    if (NOT EXISTS "${PROJECT_SOURCE_DIR}/googletest/CMakeLists.txt")
-        message(FATAL_ERROR "Missing 'googletest' submodule! Either use 'git init submodule' and 'git update submodule' to get googletest according to the README, or deactivate unit tests with -DSQLITECPP_BUILD_TESTS=OFF")
-    endif ()
-    add_subdirectory(googletest) 
-    include_directories("${PROJECT_SOURCE_DIR}/googletest/googletest/include")
+        set(PTHREAD "-pthread")
+        set(LIB_PREFIX "lib")
+        set(LIB_SUFFIX ".a")
+    endif (MSVC)
+
+    # download googletest from the github repository
+    set(PREFIX_DIR ${CMAKE_CURRENT_BINARY_DIR}/tmp)
+    ExternalProject_Add(googletest
+        PREFIX ${PREFIX_DIR}
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        INSTALL_COMMAND ""
+        CMAKE_ARGS -DCMAKE_C_COMPILER=$ENV{CC} -DCMAKE_CXX_COMPILER=$ENV{CXX}
+    )
 
     # add the unit test executable
     add_executable(SQLiteCpp_tests ${SQLITECPP_TESTS})
-    target_link_libraries(SQLiteCpp_tests gtest_main SQLiteCpp sqlite3)
+    add_dependencies(SQLiteCpp_tests googletest)
+    set(GTEST_MAIN_OBJ ${PREFIX_DIR}/src/googletest-build/googlemock/gtest/${LIB_PREFIX}gtest_main${LIB_SUFFIX})
```

Comment:
@SRombauts 
Sorry I don't have access to a Windows C++ environment, so I don't know where these gtest binaries are being generated and what they're called there. On linux they're called `libgtest.a` and `libgtest_main.a`. Would you be able to check what's happening on Windows?